### PR TITLE
feat: NotificationEmailCard component

### PR DESCRIPTION
**Pull request by braianj**
Screenshot 2024-05-16 at 4 13 25 PM
Screenshot 2024-05-16 at 4 13 56 PM
Screenshot 2024-05-16 at 4 14 04 PM
Screenshot 2024-05-16 at 4 14 15 PM
Screenshot 2024-05-16 at 4 21 24 PM
Screenshot 2024-05-16 at 4 22 03 PM

### CODE REVIEW

File: src/components/Notifications/NotificationGroupCard/NotificationGroupCard.tsx [hunk 1]
```diff
@@ -60,40 +59,34 @@ function NotificationGroupCard(props: Props) {
...
-                  onChange={(_, checked) => handleOnChangeNotificationSetting(checked, type)}
-                  checked={subscriptionDetails.messageType[toCamel(type)].email}
-                  disabled={isSavingSubscription}
-                  data-testid={NOTIFICATION_CARD_SWITCH_TEST_ID}
-                />
-              )}
-            </NotificationItemContainer>
-          ))}
-        </AccordionDetails>
+        {!isLoading && (
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationGroupCard/NotificationGroupCard.tsx:70]: Won't we be showing any kind of loader?

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        <InputContainer>
+          <TextField
+            type="email"
+            placeholder={t('settings.notifications.email.placeholder')}
+            value={email}
+            onChange={handleOnChangeEmail}
+            variant="outlined"
+            error={!isValidEmail}
+            helperText={!isValidEmail && t('settings.notifications.email.invalid_email')}
+            disabled={isLoading}
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx:119]: It seems that there's no distinction between `isSavingEmail` or `isLoading` in terms of usability. What if we simplify the component by only receiving `isLoading`?
The `isLoading` prop can contain the `isSavingEmail` prop in its logic.

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      hasConfirmEmail={false}
+      isIgnoringAllEmail={true}
+      error={null}
+      email=""
+      {...props}
+    />
+  )
+
+describe('when the component is loading', () => {
+  it('should not render the TextField component in it with disabled prop', () => {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx:32]:
```suggestion
  it('should render the TextField component in it with disabled prop', () => {
```

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    const { getByTestId } = renderNotificationEmailCard({ isLoading: true })
+    expect(getByTestId(NOTIFICATION_EMAIL_CARD_INPUT_TEST_ID).querySelector('input')).toBeDisabled()
+  })
+  it('should not render the Button component in it with disabled prop and with a CircularProgress component in it', () => {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx:36]:
```suggestion
  it('should render the Button component in it with disabled prop and with a CircularProgress component in it', () => {
```

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      )
+    })
+
+    it(`should render the Button component with the label ${t('settings.notifications.email.button_resend_label')}`, () => {
+      const { getByTestId } = renderNotificationEmailCard({ unconfirmedEmail: email })
+      expect(getByTestId(NOTIFICATION_EMAIL_CARD_BUTTON_TEST_ID).textContent).toBe(t('settings.notifications.email.button_resend_label'))
+    })
+  })
+
+  describe('and the email is has been approved', () => {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx:130]:
```suggestion
  describe('and the email has been approved', () => {
```

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    })
+    it(`should render the SpanUnconfirmedEmail component in it with the text ${t('settings.notifications.email.confirmed')}`, () => {
+      const { getByTestId } = renderNotificationEmailCard({ hasConfirmEmail: true })
+      expect(getByTestId(NOTIFICATION_EMAIL_CARD_UNCONFIRMED_TEST_ID).textContent).toBe(t('settings.notifications.email.confirmed'))
+    })
+    it('should render the Switch component in it', () => {
+      const { getByTestId } = renderNotificationEmailCard({ email })
+      expect(getByTestId(NOTIFICATION_EMAIL_CARD_SWITCH_TEST_ID)).toBeInTheDocument()
+    })
+    describe('and the user had configured to ignore all email notifications.', () => {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx:143]:
```suggestion
    describe('and the user has configured to ignore all email notifications.', () => {
```

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+      it('should set the Switch component as unchecked', () => {
+        const { getByTestId } = renderNotificationEmailCard({ email, isIgnoringAllEmail: true })
+        expect(getByTestId(NOTIFICATION_EMAIL_CARD_SWITCH_TEST_ID).querySelector('input')).not.toBeChecked()
+      })
+    })
+    describe('and the user had configured to accept all email notifications.', () => {
```
Comment (LautaroPetaccio) [inline src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx:149]:
```suggestion
    describe('and the user has configured to accept all email notifications.', () => {
```

**Review by LautaroPetaccio — APPROVED** [review]

File: src/components/Notifications/NotificationGroupCard/NotificationGroupCard.tsx (hunk 1 above)
Comment (braianj) [reply src/components/Notifications/NotificationGroupCard/NotificationGroupCard.tsx:70]: Yes but not for the text. Only disabling input and loading button

File: src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx (hunk 2 above)
Comment (braianj) [reply src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx:119]: You are right. I change the loading behaviour and forgot that `isSavingEmail`

**Pull request merged**